### ISSUE TITLE
Double-click to expand/collapse nested compositions

### ISSUE THREAD

**Issue by jminor** [OpenTimelineIO/raven#72]
`otioview` allows the user to double-click on a nested composition (e.g. a Track or Stack within a Track) to drill into it in a separate tab. It would be great if Raven had the same feature and/or a way to expand a nested composition in-place within the main window.

For the first approach, it might be best to implement #64 first and then come back to this.

For the second in-line approach, we can look at how NLEs like Avid Media Composer allow the user to expand/collapse nested compositions within the timeline window.

**Comment by jminor** [comment]
This will be easier to implement after https://github.com/OpenTimelineIO/raven/issues/72 is complete.

**Comment by jminor** [comment]
I wonder if this could work in conjunction with https://github.com/OpenTimelineIO/raven/pull/100 ? That tree view lets you see and navigate the nested composition structure - so maybe adding a control in there to set which level of composition is displayed in the main timeline view?

Right now the main timeline view assumes the top-level structure of Timeline -> Stack -> Track(s) but nested compositions could be any combination of Stack or Track parent with any Composable children. Also many of the other inspectors assume that objects are rooted in a top-level object, for example when displaying a "Global Time" or moving the playhead.

**Comment by ThomasWilshaw** [comment]
A few thoughts, does this want to open in a new tab (after implementing #64) or in the same window as the parent object? Persoanlly I feel nested compositions should be in the same window and new otio files should be inseperate tabs. If it is in the same window one idea might be to replace `appState.root` with a `std::vector<otio::SerializableObjectWithMetadata*>`, that way it can be used as a stack and moving up/down a nested layer is simply pushing or popping new roots on/off the stack. It might also be the case that some of the members of `AppState` need to be wrapped up into a new struct that could also go on the stack. We could then simple call `LoadRoot()` with the top of the stack.

I guess there would a need to be some visual queue that we are in a nested composition, a label or coloured border or something as well as buttons for moving back up to the previous level. Likewise clips in the parent timeline that have nested compositions inside them should probably be coloured differently (if they aren't already?).

How would we open a nested composition? Double click? Enter key? Right click->Open Nested? OTIOview lets you either double click or hit enter so I guess copy this?

Sorry that was a bit of a brain dump.

**Comment by jminor** [comment]
Yes, I agree that nested compositions would ideally be in the same tab. Avid Media Composer does this fairly well. Let's look at screenshots of some other applications for inspiration.

**Comment by TrevorAyl** [comment]
Source OTIO file:
[annotationreview.otio.zip](https://github.com/user-attachments/files/25795247/annotationreview.otio.zip)

**Adobe Premiere Pro**

Imports as sequences that are nested into a master sequence (i.e. PP has the nested sequences as separate elements in the project)

Adobe Premiere Pro Shows this as 4 tabs (i.e. double click on each nest and it opens as a tab):

<img width="2766" height="1020" alt="Image" src="https://github.com/user-attachments/assets/de96fe17-42ea-4275-a840-cb94ce05adf7" />

And you can switch to view the tab for each nest: (note the danger stripes indicating that the segment in the timeline does not exist in the master clip - suspect this is because timeline element starts at 00:00:00:01 and bin element is only 00:00:00:00).

<img width="2796" height="934" alt="Image" src="https://github.com/user-attachments/assets/6f002db3-78e7-46fa-a47b-6e619f483a9f" />

**Avid Media Composer**

Acts like a 'collapsed' segment in AMC - NB this is different from a 'sub-sequence'. There is no 'nested' sequence to reference to, outside the master sequence.

With Avid Media Composer you can step into a nest... :
<img width="2796" height="934" alt="Image" src="https://github.com/user-attachments/assets/0722651c-3611-485d-8f73-bd91c96b3e8c" />

Or expand it in place to view (one at a time, for display only, nest is still in timeline) - this gives a PMM insufficient media warning, suspect it is same reason as PP shows danger stripes:

<img width="2796" height="934" alt="Image" src="https://github.com/user-attachments/assets/bd36bf5b-42e2-468f-8606-7fb7406f6fbe" />

**DaVinci Resolve**

Like the other 2 shows a master timeline of 3 nests and, like Premiere Pro, shows the nested sequences in the project window as 'compound clips' (but does not list the constituent, offline, media elements):

<img width="2796" height="1688" alt="Image" src="https://github.com/user-attachments/assets/1b5061cc-a8af-4b5b-b470-1afe980bf0b3" />

You can 'decompose in place' each / all of nests:
<img width="2796" height="1688" alt="Image" src="https://github.com/user-attachments/assets/576824c1-3396-485f-a77b-ab1cfa91e9b0" />

Or open each nest (known as compound clip in DVR) in the timeline (so leave the master timeline nested):
<img width="2796" height="1688" alt="Image" src="https://github.com/user-attachments/assets/01070706-de5d-4a1c-88ab-7871640a24c5" />

**Comment by jminor** [comment]
Thanks @TrevorAyl !

Can you clarify which of these allows multiple references into a shared sub-sequence? Currently OTIO treats each nested composition as an independent copy. We should have Raven show clearly the distinction in case we add the ability to reference a shared composition in OTIO later.

**Comment by NicTanghe** [comment]
Hello I was thinking I'd make a couple of versions of a better way of visually displaying nested clips ects.
I'm not super comfortable with git however and am under the impression that as long as my previous pull request is not merged,
I'm basically unable to create a new one ?
Also, it would be unwieldy to work on a codebase that I cant run locally.
